code-analysis/crates/solidity/outputs/rust/lib/src/generated/language.rs

```diff
@@ -19,1041 +23,1087 @@ impl Language {
     }
 
     // «YulKeyword» = «BooleanLiteral» | 'break' | 'case' | 'continue' | 'default' | 'for' | 'function' | 'hex' | 'if' | 'leave' | 'let' | 'switch' ;
-    pub fn parse_yul_keyword(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_keyword.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_keyword(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_keyword)
     }
 
     // YulLeaveStatement = 'leave' ;
-    pub fn parse_yul_leave_statement(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_leave_statement.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_leave_statement(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_leave_statement)
     }
 
     // YulLiteral = «YulDecimalNumberLiteral» | «YulHexLiteral» | «AsciiStringLiteral» | «BooleanLiteral» | «HexStringLiteral» ;
-    pub fn parse_yul_literal(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_literal.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_literal(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_literal)
     }
 
     // YulStatement = YulBlock | YulVariableDeclaration | YulFunctionDefinition | YulAssignmentStatement | YulFunctionCall | YulIfStatement | YulForStatement | YulSwitchStatement | YulLeaveStatement | YulBreakStatement | YulContinueStatement ;
-    pub fn parse_yul_statement(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_statement.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_statement(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_statement)
     }
 
     // YulSwitchStatement = 'switch' YulExpression 1…*{ ( 'case' YulLiteral | 'default' ) YulBlock } ;
-    pub fn parse_yul_switch_statement(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_switch_statement.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_switch_statement(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_switch_statement)
     }
 
     // YulVariableDeclaration = 'let' YulIdentifierPath  { ',' YulIdentifierPath } [ ':=' YulExpression ] ;
-    pub fn parse_yul_variable_declaration(&self, source: &str) -> Rc<cst::Node> {
-        let (node, _errs) = self.parsers.yul_variable_declaration.parse_recovery(source);
-        node.unwrap()
+    pub fn parse_yul_variable_declaration(&self, source: &str) -> ParserOutput {
+        ParserOutput::new(source, &self.parsers.yul_variable_declaration)
+    }
+
+    pub fn parse(&self, parser_name: &str, source: &str) -> Option<ParserOutput> {
+        match parser_name {
+            "abi_coder_pragma_specifier" => Some(self.parse_abi_coder_pragma_specifier(source)),
```

Comment:
Thinking out loud: since this is now part of the public API, should we use the `RuleKind` name here (PascalCase), instead of the Rust method name (snake_case)?

1. This matches the grammar/manifest, the doc comments, the EBNF rule names in the spec, etc...
2. The snake_case only makes sense for Rust, but not other contexts (like TypeScript).

Reply:
Absolutely. I initially did that, but then changed it to match what you already had. I was going to raise this with you later :)

Reply:
Sounds good to me. I had it in `snake_case` out of convenience (as it matched the rust module names), but if it is getting out to the public API, we should probably standardize.